crates/burn-dataset/src/transform/random.rs

```diff
@@ -10,18 +10,29 @@ pub struct ShuffledDataset<D, I> {
     input: PhantomData<I>,
 }
 
+/// Generates a shuffled vector of indices up to a size.
```

Comment:
The `ShuffleDataset` could have a `SelectionDataset` internaly to avoid implementing the same logic twice while being compatible with the older API.